### PR TITLE
test(3010): add integration tests for dedupe

### LINKED ISSUE #3010: feat: add Integration tests for dedupe
Currently, there are no integration tests to verify if the deduplication feature is working correctly. We need to add tests that confirm the dedupe logic is functioning as intended.

### PR DESCRIPTION
**Pull request by dekkku**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #3010
/claim #3010

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Comment by tusharmath** [comment]
cc: @laststylebender14

File: tests/core/model.rs [hunk 1]
```diff
@@ -28,6 +28,10 @@ mod default {
         1
     }
 
+    pub fn concurrent() -> usize {
```
Comment (laststylebender14) [inline tests/core/model.rs:31]:
```suggestion
    pub fn concurrency() -> usize {
```

File: tests/core/model.rs [hunk 2]
```diff
@@ -57,6 +63,8 @@ pub struct APIRequest {
     pub test_traces: bool,
     #[serde(default)]
     pub test_metrics: bool,
+    #[serde(default = "default::concurrent")]
```
Comment (laststylebender14) [inline tests/core/model.rs:66]:
```suggestion
    #[serde(default = "default::concurrency")]
```

File: tests/execution/test-dedupe.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+type Query {
+  posts: [Post] @http(url: "http://jsonplaceholder.typicode.com/posts?id=1", dedupe: true)
+}
+
+type Post {
+  id: Int
+  title: String
+  body: String
+  userId: Int!
```
Comment (laststylebender14) [inline tests/execution/test-dedupe.md:16]: can we add more complex use case?
something like.

```graphql
type Query {
  posts: [Post] @http(url: "http://jsonplaceholder.typicode.com/posts?id=1", dedupe: true)
  user: [User] @http(url: "http://jsonplaceholder.typicode.com/users?id=1&id=2", dedupe: true)
}

type Post {
  id: Int
  title: String
  body: String
  userId: Int!
  user: User
    @http(
      url: "http://jsonplaceholder.typicode.com/users"
      query: [{key: "id", value: "{{.value.userId}}"}]
      batchKey: ["id"]
      dedupe: true
    )
}

type User {
  id: Int
  name: String
}
```

File: tests/execution/test-dedupe.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - id: 2
+        userId: 2
+```
+
+```yml @test
+- method: POST
+  url: http://localhost:8080/graphql
+  concurrency: 10
+  body:
+    query: query { posts { id, userId } }
```
Comment (laststylebender14) [inline tests/execution/test-dedupe.md:45]:
```graphql
    query: query { posts { id, userId user { id name } } user { id name } }
```

File: tests/execution/test-dedupe.md (hunk 3 above)
Comment (laststylebender14) [reply tests/execution/test-dedupe.md:16]: ensure that `user` has expectedHits set 1 to make sure dedupe is working correctly.

**Review by laststylebender14 — APPROVED** [review]

**Pull request merged**